### ISSUE TITLE
Implicits error ("Implicit annotation of `M' is not compatible") on implicits-free code

### ISSUE THREAD

**Issue by yallop** [ocamllabs/ocaml-modular-implicits#29]
This code (`bug.ml`):

``` ocaml
module type T = sig end

module Make (M : T) : sig
  module M : T
end with module M = M =
struct
  module M = M
end
```

leads to this error:

```
$ ocamlc bug.ml
File "bug.ml", line 6, characters 0-25:
Error: Signature mismatch:
       Modules do not match:
         sig module M : sig  end end
       is not included in
         sig implicit module M : sig  end end
       Implicit annotation of `M' is not compatible: it is non implicit while it was expected to be directly implicit
       File "bug.ml", line 7, characters 9-14: Actual declaration
```

This is preventing camlp4 from building, so it'd be nice to have it fixed.

**Comment by yallop** [comment]
The culprit is 65019d6, which seems to add all functor arguments to the implicit environment, not just arguments of implicit functors.

**Comment by yallop** [comment]
Fixed in ef23094e.
